test/Database_Tests/src/Database_Spec.enso

```diff
@@ -0,0 +1,358 @@
+            r_1.at 'Id' . to_vector . should_equal [4,2,3,1,6,5]
+
+        Test.specify 'should return dataflow error when passed a non-existent column' <|
+            r = df.sort by='foobar'
+            r.should_fail_with No_Such_Column_Error
+
+        Test.specify 'should correctly reorder all kinds of columns and leave the original columns untouched' <|
+            df = make_table "T8" ['ord', 'ints', 'reals', 'bools', 'texts'] ['INTEGER', 'INTEGER', 'DECIMAL', 'BOOLEAN', 'VARCHAR']
+            r = df.sort by='ord'
+            df.insert [0, 1, 1.3, False, "foo"]
+            df.insert [3, 2, 4.6, False, "foo"]
+            df.insert [2, 3, 3.2, True, "bar"]
+            df.insert [4, 4, 5.2, True, "baz"]
+            df.insert [1, 5, 1.6, False, "spam"]
+
+            ints = [1, 2, 3, 4, 5]
+            reals = [1.3, 4.6, 3.2, 5.2, 1.6]
+            bools = [False, False, True, True, False]
+            texts = ["foo", "foo", "bar", "baz", "spam"]
+
+            r.at 'ints' . to_vector . should_equal [1, 5, 3, 2, 4]
+            df.at 'ints' . to_vector . should_equal ints
+
+            r.at 'reals' . to_vector . should_equal [1.3, 1.6, 3.2, 4.6, 5.2]
+            df.at 'reals' . to_vector . should_equal reals
+
+            r.at 'bools' . to_vector . should_equal [False, False, True, False, True]
+            df.at 'bools' . to_vector . should_equal bools
+
+            r.at 'texts' . to_vector . should_equal ['foo', 'spam', 'bar', 'foo', 'baz']
+            df.at 'texts' . to_vector . should_equal texts
+
+        Test.specify 'should sort columns with specified ordering and missing placement' <|
+            c = df.at 'Rating'
+
+            r_1 = c.sort
+            r_1.to_vector.should_equal [2.2, 3.0, 3.0, 7.3, Nothing, Nothing]
+
+            r_2 = c.sort order=Sort_Order.Descending
+            r_2.to_vector.should_equal [7.3, 3.0, 3.0, 2.2, Nothing, Nothing]
+
+            r_3 = c.sort order=Sort_Order.Descending missing_last=False
+            r_3.to_vector.should_equal [Nothing, Nothing, 7.3, 3.0, 3.0, 2.2]
+
+
+sqlite_specific_spec connection =
+    Test.group "Error Handling" <|
+        Test.specify "should wrap errors" <|
+            connection.execute_query "foobar" . should_fail_with Sql_Error
+            connection.execute_update "foobar" . should_fail_with Sql_Error
+
+            action = connection.execute_query "SELECT A FROM undefined_table"
+            action . should_fail_with Sql_Error
+            action.catch.to_text . should_equal "[SQLITE_ERROR] SQL error or missing database (no such table: undefined_table)"
+
+sqlite_spec =
+    file = File.new "target/sqlite_test.db"
+    file.delete_if_exists
+    connection = Database.open_sqlite_file file
+    here.spec connection
+    here.sqlite_specific_spec connection
+    connection.close
+    file.delete
+
+postgres_spec =
+    # TODO [RW] use env vars to read tmp DB config
+    IO.println "PostgreSQL test database is not configured, skipping."
```

Comment:
Use the pending group feature after you rebase on top of a main that contains my last PR (Python)